### ISSUE TITLE
ALLEGRO_FULLSCREEN_WINDOW has UI calls on a background thread

### ISSUE THREAD

**Issue by SiegeLord** [liballeg/allegro5#876]
See https://www.allegro.cc/forums/thread/617208/1034566#target

https://github.com/liballeg/allegro5/blob/da4b7162941b8f364a02bba85b5c2a1b168f5478/src/macosx/osxgl.m#L1578

**Comment by joecarl** [comment]
I've also seen this error appearing with ALLEGRO_WINDOWED (High sierra Xcode 9). Is there any chance of fixing this in the next release?

**Comment by pedro-w** [comment]
I've started to look at this. Seems it would be straightforward if we use the [Dispatch Queue](https://developer.apple.com/documentation/dispatch/dispatch_queue?language=objc) API. This requires OSX 10.6 or later - assume that's OK (Snow Leopard, released nearly a decade ago)
There are more instances of this (e.g. `_al_osx_mouse_was_installed`) and the main trouble will be trying to find them all, since it's not obvious just from viewing the code which thread it's called from.
Possibly related, #936.

**Comment by pedro-w** [comment]
In other places we use `performSelectorOnMainThread:withObject:waitUntilDone:` with an Allegro specific helper object to achieve the same thing, and I am not proposing to change these, even though it would probably make the code a bit neater.
e.g.
https://github.com/liballeg/allegro5/blob/fe8039233112ad26451480e57038b571dd407402/src/macosx/hidjoy-10.4.m#L373-L375
However I will if people think it's worth it.

**Comment by SiegeLord** [comment]
Yeah, we can easily up the requirement to way more than 10.6. It'd be great to just get this to be correct, but if we can clean the code up that'd be amazing too.

**Comment by pedro-w** [comment]
OK, will do.
